### PR TITLE
NIFI-7321 - Allow NiFi admins to configure whether Jetty will send th…

### PR DESCRIPTION
**Pull request by thenatog**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by alopresto** [comment]
Please add the new property and default value to the template `nifi.properties` file. See my commit on NIFIREG-380 for an example.

**Comment by thenatog** [comment]
Added the new property to the default template nifi.properties

**Comment by alopresto** [comment]
Reviewing...

**Comment by alopresto** [comment]
I attempted to verify this behavior by setting `nifi.web.should.send.server.version=false` in `nifi.properties` and running the application. Most requests, including 404 errors, did not return the Jetty server version in the response headers, but some did (Help > `overview.html`, for example). We should check that all web contexts are treated the same here. 
NiFi in browser with developer tools showing Jetty server version in response header

**Comment by thenatog** [comment]
Added a small change and I'm seeing this:
Screen Shot 2020-04-28 at 2 47 15 PM

Committing this update momentarily.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java [hunk 1]
```diff
@@ -908,6 +909,7 @@ private ServerConnector createUnconfiguredSslServerConnector(Server server, Http
         final HttpConfiguration httpsConfiguration = new HttpConfiguration(httpConfiguration);
         httpsConfiguration.setSecureScheme("https");
         httpsConfiguration.setSecurePort(port);
+        httpConfiguration.setSendServerVersion(props.shouldSendServerVersion());
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java:912]: Why is this variable `httpConfiguration` vs. `httpsConfiguration`?

Comment (thenatog) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java]: Fixed

**Pull request closed without merging**